lntest/itest/lnd_amp_test.go

```diff
@@ -38,6 +38,15 @@ func testSendPaymentAMPInvoiceCase(net *lntest.NetworkHarness, t *harnessTest,
 	ctx := newMppTestContext(t, net)
 	defer ctx.shutdownNodes()
 
+	// Subscribe to bob's invoices. Do this early in the test to make sure
+	// that the subscription has actually been completed when we add an
+	// invoice. Otherwise the notification will be missed.
+	req := &lnrpc.InvoiceSubscription{}
+	ctxc, cancelSubscription := context.WithTimeout(ctxb, defaultTimeout)
```

Comment:
Switching this from `WithCancel` to `WithTimeout` will change the behavior. Can we be certain that the default timeout is enough for the duration that `bobInvoiceSubscription` will be active for?

Reply:
That is a good point. Maybe not. I reverted that part of the change to not break anything.